### ISSUE TITLE
Invalid value reported when there is no value

### ISSUE THREAD

**Issue by nemonemi** [date-fns/date-fns#2829]
Right now, when a field contains no values the "value" is reported as an `Invalid value` - date object.
Is there a utility method to differentiate between invalid content and no content?

**Comment by fturmel** [comment]
Could you specify which `date-fns` function you're using, and share a short code example? I'm not 100% sure what you're asking for.

**Comment by nemonemi** [comment]
I am using react-hook-form. I've set the field to be a date field, meaning the library automatically transforms the value to be a Date object. 
To validate that value I'm using their `validate` method that receives the date object as a value.
I could turn off the automatic formatting, but that would mean more manual formatting on my side, which is not desired.


I'd like to know if date-fns has any way of differentiating between an invalid date value, and an empty date value (empty input field).

**Comment by fturmel** [comment]
It depends on the specifics, but you could try the `isValid` function maybe? Look at the truth table in the docs and see if that fits your need: https://date-fns.org/docs/isValid

If you're trying to determine if a JS Date instance is valid or not, you can also simply use `isNaN(new Date('bad date').getTime()) // true`

> empty date value (empty input field)

It needs to be more specific than that, because the Date constructor is pretty accepting in terms of inputs and can sometimes behave differently based on the JS runtime.

```js
// valid date
new Date()
new Date(0)
new Date(null)

// invalid date
new Date('')
new Date(undefined)

// valid in V8 (Chrome), invalid date in JavaScriptCore (Safari)
new Date('2021-12-08 00:00:00.000Z')
```

Hoping that helps!

**Comment by nemonemi** [comment]
Thanks for the super quick responses @fturmel!

From what I saw in the docs and your example (thanks btw) it is not possible to deduce if the value is an empty string from an invalid value.

```ts
isNaN(new Date('').getTime()) // true
isNaN(new Date('bad date').getTime()) // true
```
The same goes for the `isValid` method.

Thus, I had to change the field to string and handle the date validation and conversion manually.

I'll close it for now and hopefully, this helps others searching for the same issue.
